lib/services/debug-service.ts

```diff
@@ -0,0 +1,65 @@
+import { platform } from "os";
+import { EventEmitter } from "events";
+import { CONNECTION_ERROR_EVENT_NAME } from "../constants";
+
+class DebugService extends EventEmitter {
+	constructor(private $devicesService: Mobile.IDevicesService,
+		private $androidDebugService: IPlatformDebugService,
+		private $iOSDebugService: IPlatformDebugService,
+		private $errors: IErrors,
+		private $hostInfo: IHostInfo,
+		private $devicePlatformsConstants: Mobile.IDevicePlatformsConstants) {
+		super();
+	}
+
+	public async debug(debugData: IDebugData, options: IDebugOptions): Promise<string> {
+		const device = this.$devicesService.getDeviceByIdentifier(debugData.deviceIdentifier);
+		const debugService = this.getDebugService(device);
+
+		debugService.on(CONNECTION_ERROR_EVENT_NAME, (e: Error) => this.emit(CONNECTION_ERROR_EVENT_NAME, e));
```

Comment:
who will remove this handler? In case you call `debug` method three times, you'll have three handlers for the error event. So in case an error is raised, the one who listens for DebugService's `CONNECTION_ERROR_EVENT_NAME` event will receive three errors.